dist/rules/no-internal.js

```diff
@@ -168,36 +168,40 @@ module.exports = {
         if (jsDoc.tags) {
           for (const tag of jsDoc.tags) {
             if (bannedTags.includes(tag.tagName.escapedText) && isCheckedFile(declaration)) {
-              let name;
-              if (declaration.kind === ts.SyntaxKind.Constructor)
-                name = declaration.parent.symbol.escapedName;
-              else {
-                name = declaration.symbol.escapedName;
-                const parentSymbol = getParentSymbolName(declaration);
-                if (parentSymbol)
-                  name = `${parentSymbol}.${name}`;
-              }
-
-              context.report({
-                node,
-                messageId: "forbidden",
-                data: {
-                  kind: syntaxKindFriendlyNames.hasOwnProperty(declaration.kind) ? syntaxKindFriendlyNames[declaration.kind] : "unknown object type " + declaration.kind,
-                  name,
-                  tag: tag.tagName.escapedText,
+              //Violation key to track and report violations on a per-usage basis
+              const violationKey = `${declaration.kind}_${declaration.symbol.escapedName}_${tag}_${node.range[0]}`;
```

Comment:
I'm OK with this fix for now, but really can't we reduce this rule to checking the identifier leaf nodes anyway, if they're a reference? I feel like it could be simpler. I wonder why @paulius-valiunas didn't go with that approach originally.

Other than that as a stopgap I'm willing to approve this PR once you respond to this comment with your opinion.

Reply:
I believe we check the other node types to handle cases where JSDoc tags are attached directly to function calls, constructor calls, and class member accesses.

Reply:
you can't attach JSDoc tags to usages though, only definitions. So I'm not sure what you mean. A constructor call still contains an identifier leaf node. So do function calls, member access, etc. Try typing any of the syntax you mentioned into https://astexplorer.net/ and you'll notice there's always an "Identifier" node somewhere, of which we can ask typescript for the definition's JSDoc tags.

Reply:
Ah yes, sorry got confused. Yeah, I think checking just the identifier leaf nodes would be simpler.

Reply:
do you want to defer that work and merge this PR for now?

Reply:
Yes, I think it's good to get this fix in first.

Reply:
Tbh though that will also fix this. And now that I think about it, do the existing matchers even cover the importing of an https://github.com/internal let binding? They're rare but they don't seem to...

Reply:
after poking at it myself (and cross referencing with the [deprecation](https://github.com/gund/eslint-plugin-deprecation/blob/master/src/rules/deprecation.ts) rule, it seems that the typescript API doesn't really work that way, and this all seems the right way to do it for the most part. Resolving and approving